engines/director/sprite.cpp

```diff
@@ -391,6 +392,14 @@ void Sprite::setPattern(uint16 pattern) {
 	}
 }
 
+void Sprite::setAutoPuppet(AutoPuppetProperty property, bool value) {
+	setFlagBit<uint32>(&_autoPuppet, property, value);
+}
+
+bool Sprite::getAutoPuppet(AutoPuppetProperty property) {
```

Comment:
Where do you use this? The current code has checks against `_autoPuppet`  in frame.cpp

Reply:
Well, they need to be rewritten, because now there is not a single _autoPuppet that can be used to check if sprite is autoPuppeted, each property of sprite is autoPuppet so depending upon if that specific property is auto-puppeted (for example this `Sprite::getAutoPuppet` return true, then we skip that change).

I will do one thing, make the required changes in frame loading as well to implement individual autoPuppets.